### DISCUSSION TITLE
./gradlew build error github action

### DISCUSSION THREAD

**Discussion by Zaina-Khan** [community/community#26445] [Actions]
<p>I am trying to build my app on actions<br>
./gradlew: line 1: ?xml: No such file or directory<br>
./gradlew: line 2: androidx.coordinatorlayout.widget.CoordinatorLayout: No such file or directory<br>
./gradlew: line 3: $‘android:layout_width=match_parent\r’: command not found<br>
./gradlew: line 4: $‘android:layout_height=match_parent\r’: command not found<br>
./gradlew: line 5: xmlns:app=<a href="http://schemas.android.com/apk/res-auto" rel="nofollow noopener">http://schemas.android.com/apk/res-auto</a><br>
: No such file or directory<br>
./gradlew: line 6: android:fitsSystemWindows=true: command not found<br>
./gradlew: line 7: $’\r’: command not found<br>
./gradlew: line 8: com.google.android.material.appbar.AppBarLayout<br>
: No such file or directory<br>
./gradlew: line 9: android:id=@+id/appBar<br>
: No such file or directory<br>
./gradlew: line 10: $‘android:layout_width=match_parent\r’: command not found<br>
./gradlew: line 11: android:layout_height=wrap_content: command not found<br>
./gradlew: line 12: $’\r’: command not found<br>
./gradlew: line 13: com.google.android.material.appbar.MaterialToolbar<br>
: No such file or directory<br>
./gradlew: line 14: android:id=@+id/toolbar<br>
: No such file or directory<br>
./gradlew: line 15: $‘android:layout_width=match_parent\r’: command not found<br>
./gradlew: line 16: android:layout_height=?attr/actionBarSize/: No such file or directory<br>
./gradlew: line 17: $’\r’: command not found<br>
./gradlew: line 18: $’\r’: command not found<br>
./gradlew: line 19: /com.google.android.material.appbar.AppBarLayout: No such file or directory<br>
./gradlew: line 20: $’\r’: command not found<br>
./gradlew: line 21: syntax error near unexpected token <code>newline'   ./gradlew: line 21: </code>&lt;/androidx.coordinatorlayout.widget.CoordinatorLayout&gt;’<br>
##[error]Process completed with exit code 2.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/zaina-khan">@zaina-khan</a> , </p>
<p>For me to reproduce, better understand your scenario, would you mind sharing your workflow yml file and a sample project here? </p>

**Comment by Zaina-Khan** [comment]
<p>Sure here<br>
Worflow</p>
<pre><code>name: Build &amp; Publish Debug APK
on:
 push:
  branches:
   - master
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - name: set up JDK 1.8
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
    - name: Make Gradle executable
      run: chmod +x ./gradlew
    - name: Build with Gradle
      run: ./gradlew build
    - name: Build Debug APK
      run: ./gradlew assembleDebug
</code></pre>
<p>Any project I create on Android studio in windows, gives me this error. The current one I first encountered with is currently private. You can reproduce this with creating any project on windows. If it doesn’t I can put a repo on git encountering this. Thanks!</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/zaina-khan">@zaina-khan</a> , </p>
<p>I could not reproduce your issue in my side, could you please share a sample git repo here? </p>
<p>What’s the result if you change to use windows runner? </p>
<pre><code>runs-on: windows-latest
</code></pre>
<p>I created a new android project with Basic Activity template, and then push the code to my github repo. I also used your workflow, it built successfully. <a href="https://github.com/zhuorg/AndroidApp/runs/625381823?check_suite_focus=true" rel="nofollow noopener">https://github.com/zhuorg/AndroidApp/runs/625381823?check_suite_focus=true </a>  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2b34a7d27d1e0865288b6825220cf4173a276db5.png" data-download-href="/uploads/short-url/6adlp5ANICBfCLTalM4MDetxw5T.png?dl=1" title="new android app.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181097853-e105771e-4b90-4d12-85a8-cbff219a0cc2.png" alt="new android app.png" data-base62-sha1="6adlp5ANICBfCLTalM4MDetxw5T" width="658" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b34a7d27d1e0865288b6825220cf4173a276db5_2_658x500.png, https://aws1.discourse-cdn.com/github/original/2X/2/2b34a7d27d1e0865288b6825220cf4173a276db5.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2b34a7d27d1e0865288b6825220cf4173a276db5.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2b34a7d27d1e0865288b6825220cf4173a276db5_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">new android app.png</span><span class="informations">898×682 28 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Zaina-Khan** [comment]
<p><a class="mention" href="/yanjingzhu">@yanjingzhu</a> Thank you so much! It works after using windows, I googled around and seems to be an issue with line endings (CRLF). I can’t seem to be able to find a fix for it, if you ccan will be helpful thanks!<br>
Because I wanted to use ubuntu as the OS.</p>
<p><a href="https://stackoverflow.com/questions/44989520/android-studio-crlf-vs-lf-for-a-git-based-multiplatform-project" rel="nofollow noopener">https://stackoverflow.com/questions/44989520/android-studio-crlf-vs-lf-for-a-git-based-multiplatform-project</a></p>
